src/engine/userapi/ProtoUserDataStream.java

```diff
@@ -1,9 +1,41 @@
 package engine.userapi;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Prototype data sent between user and controller
  * user output data
  */
 public interface ProtoUserDataStream {
-
+	
+	/**
+	 * Input data
```

Comment:
List<Integer> rectangles = data.getList(); // grab the input list
        UserDataStream inputStream = new UserDataStream(rectangles); // turn it into a user data stream
        return sysctl.receiveUserRequest(inputStream); // call the controller an set the entry point there
    } the classes are separated, the list gets transferred to the userdata stream so that it can all be contained within one class. This avoids adding extra subroutines to deal with multiple input and output sources. We can change this if you think it would be better, but I think it would be a fairly easy change later (just add a class and change a few lines of code in the sendUserRequest(...)

Reply:
Gotcha